src/Microsoft.AspNet.Http.Features/FeatureReference.cs

```diff
@@ -34,5 +34,11 @@ public T Update(IFeatureCollection features, T feature)
             _revision = features.Revision;
             return feature;
         }
+
+        public void Reset()
```

Comment:
Need to check; they are heap allocated - so would be another chunk of memory but they are only a pointer and an int...